NuRadioReco/modules/ARA/triggerSimulator.py

```diff
@@ -200,47 +247,44 @@ def run(self, evt, station, det,
+        trigger_times = []
         for channel in station.iter_channels():
             channel_id = channel.get_id()
             if channel_id not in triggered_channels:
                 continue
             trigger[channel_id] = self.has_triggered(channel)
+            if trigger[channel_id]:
+                times = channel.get_times()
+                trace_after_diode = self.tunnel_diode(channel)
+                arg_trigger = np.argwhere( trace_after_diode == np.min(trace_after_diode) )[0,0]
+                trigger_times.append(times[arg_trigger])
 
         has_triggered = False
         trigger_time_sample = None
-        # loop over the trace with a sliding window of "coinc_window"
-        coinc_window_samples = np.int(np.round(coinc_window * channel.get_sampling_rate()))
-        trace_length = len(station.get_channel(0).get_trace())
-        sampling_rate = station.get_channel(0).get_sampling_rate()
-
-        for i in range(0, trace_length - coinc_window_samples):
-            istop = i + coinc_window_samples
-            coinc = 0
-            trigger_times = []
-            for iCh, tr in trigger.items():  # loops through triggers of each channel
-                tr = np.array(tr)
-                mask_trigger_in_coind_window = (tr >= i) & (tr < istop)
-                if(np.sum(mask_trigger_in_coind_window)):
-                    coinc += 1
-                    trigger_times.append(tr[mask_trigger_in_coind_window][0])  # save time/sample of first trigger in
-
-            if coinc >= number_concidences:
+
+        trace_times = station.get_channel(0).get_times()
+        trigger_times = np.array(trigger_times)
+        slice_left = int(coinc_window/2/(trace_times[1]-trace_times[0]))
+        slice_right = len(trace_times)-slice_left
+        for trace_time in trace_times[slice_left:slice_right]:
+            if ( np.sum( np.abs(trace_time-trigger_times) <= coinc_window/2 ) >= number_concidences ):
                 has_triggered = True
-                trigger_time_sample = min(trigger_times)
+                trigger_time_sample = np.min(trigger_times)
                 break
 
         trigger = IntegratedPowerTrigger(trigger_name, power_threshold,
                                          coinc_window, channels=triggered_channels,
                                          number_of_coincidences=number_concidences,
                                          power_mean=self._power_mean, power_std=self._power_std)
+
         if not has_triggered:
             trigger.set_triggered(False)
             logger.info("Station has NOT passed trigger")
             trigger_time_sample = 0
-            station.get_trigger().set_trigger_time(trigger_time_sample / sampling_rate)
+            trigger.set_trigger_time(trigger_time_sample)
         else:
             trigger.set_triggered(True)
-            trigger.set_trigger_time(trigger_time_sample / sampling_rate)
+            trigger.set_trigger_time(trigger_time_sample)
```

Comment:
you might want to rename the variable from `trigger_time_sample` to `trigger_time`. Otherwise its confusing.

Reply:
Changed.